### PR TITLE
Added rubocop / codeclimate config and fixed current offenses

### PR DESCRIPTION
**Pull request by cristianbica**
Better to write code directly respecting [rails' style guide](https://github.com/rails/rails/blob/master/.rubocop.yml) from early time so we save us some trouble when merging into rails.
Did a slight change in the `.rubocop.yml` by targeting ruby 2.3 rather than ruby 2.2 because the safe navigation operator is used.

Should we setup codeclimate for this repo?

### CODE REVIEW

File: lib/active_storage/blob.rb [hunk 1]
```diff
@@ -48,7 +48,6 @@ def url_for_direct_upload(expires_in: 5.minutes)
     service.url_for_direct_upload key, expires_in: expires_in, content_type: content_type, content_length: byte_size
   end
 
-
```
Comment (dhh) [inline lib/active_storage/blob.rb]: These are all intentional to group related method sets together. Whatever rule is creating this as a violation, let's drop it.

File: test/attachments_test.rb [hunk 2]
```diff
@@ -56,7 +56,6 @@ class ActiveStorage::AttachmentsTest < ActiveSupport::TestCase
     end
   end
 
-
```
Comment (dhh) [inline test/attachments_test.rb]: Intentional.

**Comment by dhh** [comment]
👍

**Comment by cristianbica** [comment]
@dhh done

File: test/attachments_test.rb [hunk 3]
```diff
@@ -66,8 +66,8 @@ class ActiveStorage::AttachmentsTest < ActiveSupport::TestCase
 
   test "attach new blobs" do
     @user.highlights.attach(
-      { io: StringIO.new("STUFF"), filename: "town.jpg", content_type: "image/jpg" }, 
-      { io: StringIO.new("IT"), filename: "country.jpg", content_type: "image/jpg" })
+      { io: StringIO.new("STUFF"), filename: "town.jpg", content_type: "image/jpg" },
+      io: StringIO.new("IT"), filename: "country.jpg", content_type: "image/jpg")
```
Comment (dhh) [inline test/attachments_test.rb]: This one isn't quite right either. I know that rubucop is trying to enforce here, but it doesn't make sense in this instance. Don't think whatever rule is being enforced here is valuable enough to keep.

File: test/test_helper.rb [hunk 4]
```diff
@@ -15,7 +15,6 @@
   {}
 end
 
-
```
Comment (dhh) [inline test/test_helper.rb]: This is also intentional. Wonder why that slipped through?

Comment (cristianbica) [inline test/test_helper.rb]: let me run the auto-correct thing again on master with the updated rules

File: test/attachments_test.rb (hunk 3 above)
Comment (cristianbica) [inline test/attachments_test.rb]: Actually the rule has a `context_dependent` config which enforces braces in this case:

```
# bad
some_method(x, y, {a: 1, b: 2})
some_method(x, y, {a: 1, b: 2}, a: 1, b: 2)

# good
some_method(x, y, a: 1, b: 2)
some_method(x, y, {a: 1, b: 2}, {a: 1, b: 2})
```

**Comment by cristianbica** [comment]
@dhh really done now :) all changes are `"` instead of `'` and extra spaces

**Comment by dhh** [comment]
👍

**Pull request merged**